feat: Add a `drupal` project type that handles Drupal8-11, fixes #4957 (#5991)
ddev/ddev: docs/content/users/configuration/config.md

| Type | Default | Usage
| -- | -- | --
| :octicons-file-directory-16: project | `php` | Can be `backdrop`, `craftcms`, `django4`, `drupal6`, `drupal7`, `drupal`,  `laravel`, `magento`, `magento2`, `php`, `python`, `shopware6`, `silverstripe`, `typo3`, or `wordpress`.

The `php` and `python` types don’t attempt [CMS configuration](../../users/quickstart.md) or settings file management and can work with any project.

> The legacy `drupal8`, `drupal9`, and `drupal10` project types can also be specified, but if you just use `drupal` DDEV will autodetect the type and use the recommended settings.

Reviewer: ```suggestion
The legacy `drupal8`, `drupal9`, and `drupal10` project types can also be specified, but by using `drupal` DDEV will autodetect the type and use the recommended settings.
```

Reviewer 2: > The legacy `drupal8`, `drupal9`, and `drupal10`

There has been talk about using the terms "modern drupal" for Drupal 8+, and "legacy Drupal" for Drupal =<7. 
With that in mind, I think we should avoid using "legacy" in this doc context.

Author: Good point @tyler36 - They're "legacy" only in the sense that we're not going to use specific terms in DDEV any more.  Please suggest alternative wording, thanks.

Reviewer: How about the following?  
`The project types drupal8, drupal9, and drupal10 can still be directly specified,...`

Reviewer 2: > The project types drupal8, drupal9, and drupal10 can still be directly specified,...

@rpkoller, I like that! It keeps with the DDEV terminology.

Author: Updated to 
> The former DDEV project types `drupal8`, `drupal9`, and `drupal10` can still be manually specified; by using `drupal` instead, DDEV will autodetect the correct type and its corresponding settings.
